pymaid/upload.py

```diff
@@ -1713,25 +1742,44 @@ def add_connector(coords, remote_instance=None):
                         Use this to add treenodes.
 
     """
-    remote_instance = utils._eval_remote_instance(remote_instance)
-
+    remote_instance = utils._eval_remote_instance(remote_instance)    
+    resp = []
     if not utils._is_iterable(coords[0]):
         coords = [coords]
 
     coords = np.array(coords)
 
     if coords.shape[1] != 3:
         raise ValueError('Expected x/y/z coordinates, got {}'.format(coords.shape[1]))
+    if check_existing:
+        for coord in coords:
+            existing_connector = fetch.get_connectors_in_bbox([[coord[0], coord[0]+1],
+                                                               [coord[1], coord[1]+1],
+                                                               [coord[2], coord[2]+1]],
+                                                              ret='IDS', remote_instance=remote_instance)    
+            
+            if len(existing_connector) == 0:
+                url = [remote_instance._create_connector_url()] 
+                post = [{'pid': remote_instance.project_id, 
+                        'confidence': 5,
+                        'x': coord[0],
+                        'y': coord[1],
+                        'z': coord[2]}]
+                # TODO run one API call for all connectors that need to be uploaded, instead of
```

Comment:
The easiest way to speed this up is to first generate urls for all bounding boxes and then fetch them all in parallel.

Reply:
How would you do this? I think I previously tried this by passing lists of bounding boxes to `fetch.get_connectors_in_bbox` and that didn't work. So I guess I'd need to build a list of `post` dicts myself and do the parallel fetch within this function instead of letting `fetch.get_connectors_in_bbox` do the fetch? Or, I suppose maybe the better way to do this would be to extend `fetch.get_connectors_in_bbox`'s functionality so that it could accept lists of bboxes, do the parallel fetch, and then return a list with elements corresponding to each of the requested bboxes. What do you think?